lib/sassc/functions_handler.rb

```diff
@@ -10,25 +10,47 @@ def setup(native_options)
 
       list = Native.make_function_list(Script.custom_functions.count)
 
-      functs = FunctionWrapper.extend(Script::Functions)
-      functs.options = @options
+      functions = FunctionWrapper.extend(Script::Functions)
+      functions.options = @options
 
       Script.custom_functions.each_with_index do |custom_function, i|
-        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |s_args, cookie|
-          length = Native.list_get_length(s_args)
+        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |native_argument_list, cookie|
+          native_argument_list_length = Native.list_get_length(native_argument_list)
+          custom_function_arguments = []
+          errorTag = nil
 
-          v = Native.list_get_value(s_args, 0)
-          v = Native.string_get_value(v).dup
+          (0...native_argument_list_length).each do |i|
+            native_value = Native.list_get_value(native_argument_list, i)
 
-          s = Script::String.new(Script::String.unquote(v), Script::String.type(v))
+            case value_tag = Native.value_get_tag(native_value)
+            when :sass_null
+              # no-op
+            when :sass_string
+              native_string = Native.string_get_value(native_value)
+              argument = Script::String.new(Script::String.unquote(native_string), Script::String.type(native_string))
 
-          value = functs.send(custom_function, s)
+              custom_function_arguments << argument
+            else
+              errorTag = error(message: "Sass argument of type #{value_tag} unsupported")
+              break
+            end
+          end
 
-          if value
-            value = Script::String.new(Script::String.unquote(value.to_s), value.type)
-            value.to_native
+          if errorTag
```

Comment:
within a proc, `break` returns from the function that the proc was called in.   `next` just returns from the proc.

Since the custom function is a Proc, it is probably called and the return value is used afterwords in the C function.  By calling `break` the code after the proc doesn't get to execute, which probably caused the segfault.